### PR TITLE
Implemented global configuration for bitbucket token

### PR DESCRIPTION
**Pull request by maksudur-rahman-maruf**
- Added Bitbucket Api Token (Optional) field in the Jenkins Plugin UI.

### CODE REVIEW

File: src/main/java/com/synopsys/integration/jenkins/scan/service/scm/BitbucketRepositoryService.java [hunk 1]
```diff
@@ -36,9 +36,9 @@ public Bitbucket fetchBitbucketRepositoryDetails(Jenkins jenkins, Map<String, Ob
                 BitbucketSCMSource bitbucketSCMSource = (BitbucketSCMSource) scmSource;
 
                 String bitBucketToken = (String) scanParameters.get("bitbucket_token");
-                if(bitBucketToken == null) {
-                    bitBucketToken = SCMRepositoryService.getCredentialsToken(bitbucketSCMSource.getCredentialsId());
-                }
+//                if(bitBucketToken == null) {
```
Comment (zaman-akib) [inline src/main/java/com/synopsys/integration/jenkins/scan/service/scm/BitbucketRepositoryService.java:39]: We should add a private method like `getBitbucketTokenFromGlobalConfig()`  here at `BitbucketRepositoryService` 

```
public String getBitbucketTokenFromGlobalConfig() {
        ScannerGlobalConfig config = GlobalConfiguration.all().get(ScannerGlobalConfig.class);
        if (config != null && !Utility.isStringNullOrBlank(config.getBitbucketToken())) {
            return config.getBitbucketToken().trim();
        }
        return null;
    }
```

then 

```
if (bitbucketToken == null) {
   bitbucketToken = getBitbucketTokenFromGlobalConfig()
}
```

**_N.B. Here there may be test failure in `BitbucketRepositoryServiceTest`_**

File: src/main/java/com/synopsys/integration/jenkins/scan/service/scm/BitbucketRepositoryService.java [hunk 2]
```diff
@@ -36,9 +36,9 @@ public Bitbucket fetchBitbucketRepositoryDetails(Jenkins jenkins, Map<String, Ob
...
+//                    bitBucketToken = SCMRepositoryService.getCredentialsToken(bitbucketSCMSource.getCredentialsId());
+//                }
```
Comment (zaman-akib) [inline src/main/java/com/synopsys/integration/jenkins/scan/service/scm/BitbucketRepositoryService.java:41]: If we don't get bitbucketToken either from pipeline input or global config but prComment is set to true then, 
there should be some error for user.

File: src/main/resources/com/synopsys/integration/jenkins/scan/extension/global/ScannerGlobalConfig/config.jelly [hunk 3]
```diff
@@ -6,8 +6,11 @@
         <f:entry field="blackDuckApiToken" title="Black Duck Api Token">
             <f:password/>
         </f:entry>
-        <f:entry field="synopsysBridgeDownloadUrl" title="Synopsys Bridge Download URL">
+        <f:entry field="synopsysBridgeDownloadUrl" title="Synopsys Bridge Download URL (Optional)">
             <f:textbox/>
         </f:entry>
+        <f:entry field="bitbucketApiToken" title="Bitbucket Api Token (Optional)">
```
Comment (zaman-akib) [inline src/main/resources/com/synopsys/integration/jenkins/scan/extension/global/ScannerGlobalConfig/config.jelly:12]: `bitbucketToken`

**Review by zaman-akib — CHANGES_REQUESTED** [review]

File: src/main/java/com/synopsys/integration/jenkins/scan/service/scm/BitbucketRepositoryService.java (hunk 1 above)
Comment (maksudur-rahman-maruf) [reply src/main/java/com/synopsys/integration/jenkins/scan/service/scm/BitbucketRepositoryService.java:39]: Thanks for the code snippet.

File: src/main/java/com/synopsys/integration/jenkins/scan/service/scm/BitbucketRepositoryService.java [hunk 4]
```diff
@@ -46,10 +40,9 @@ public Bitbucket fetchBitbucketRepositoryDetails(Jenkins jenkins, Map<String, Ob
                         bitbucketToken = getBitbucketTokenFromGlobalConfig();
                     } else {
 //                        bitbucketToken = SCMRepositoryService.getCredentialsToken(bitbucketSCMSource.getCredentialsId());
-                        try {
+                        Boolean prComment = (Boolean) scanParameters.get("blackduck_automation_prcomment");
```
Comment (zaman-akib) [inline src/main/java/com/synopsys/integration/jenkins/scan/service/scm/BitbucketRepositoryService.java:43]: use `ApplicationConstants.BLACKDUCK_AUTOMATION_PRCOMMENT_KEY` 

`scanParameters.get(ApplicationConstants.BLACKDUCK_AUTOMATION_PRCOMMENT_KEY)`

File: src/main/java/com/synopsys/integration/jenkins/scan/global/ExceptionMessages.java [hunk 5]
```diff
@@ -8,4 +8,8 @@ public static String scannerFailedWithExitCode(int exitCode) {
         return "Scanner failed with exit code " + exitCode;
     }
 
+    public static String scannerFailureMessage(String message) {
+        return "Scanner failure! " + message;
```
Comment (zaman-akib) [inline src/main/java/com/synopsys/integration/jenkins/scan/global/ExceptionMessages.java:12]: first part of the message can be `Scan failure!` or `Scanner failed!`

**Review by zaman-akib — APPROVED** [review]

**Pull request merged**